src/super_gradients/training/pipelines/pipelines.py

```diff
@@ -59,6 +67,15 @@ def __init__(self, model: SgModule, image_processor: Union[Processing, List[Proc
             image_processor = ComposeProcessing(image_processor)
         self.image_processor = image_processor
 
+        self.fuse_model = fuse_model  # If True, the model will be fused in the first forward pass, to make sure it gets the right input_size
+
+    def _fuse_model(self, input_example: torch.Tensor):
+        logger.info("Fusing some of the model's layers. If this takes too much memory, you can deactivate it by setting `fuse_model=False`")
+        self.model = copy.deepcopy(self.model)
```

Comment:
We need to release GPU memory here once we are done predicting (that is, when the original model goes into train mode).
We can delete the model copy / entire pipeline and then call torch.cuda.empty_cache() (when cuda is a available of course).

Reply:
Done with:
``` python
    def train(self, mode: bool = True):
        super().train(mode)

        self._get_pipeline.cache_clear()
        torch.cuda.empty_cache()
```